core/deployment/src/main/java/io/quarkus/deployment/pkg/jar/AbstractJarBuilder.java

```diff
@@ -105,6 +106,7 @@ protected static void copyFiles(ApplicationArchive archive, ArchiveCreator archi
             Map<String, List<byte[]>> services,
             Predicate<String> ignoredEntriesPredicate) throws IOException {
         try {
+            Map<String, Path> pathsToCopy = new TreeMap<>();
```

Comment:
Not a big deal (not sure if it's relevant to this case), but you can also make it case-insensitive to avoid surprises:
```suggestion
            Map<String, Path> pathsToCopy = new TreeMap<>(String.CASE_INSENSITIVE_ORDER);
```

Reply:
We don’t really care about the ordering as long as it’s stable.